src/odata-to-abstract-sql.ts

```diff
@@ -1199,7 +1199,17 @@ export class OData2AbstractSQL {
 		} else if (prop.count) {
 			const query = new Query();
 			query.select.push(['Count', '*']);
-			this.AddNavigation(query, this.defaultResource!, prop.name);
+			const aliasedResource = this.AddNavigation(
+				query,
+				this.defaultResource!,
+				prop.name,
+			);
+			if (prop.options?.$filter) {
```

Comment:
This reads to me like previously you couldn't have done eg
```
/pilot?$filter=trained__pilot/$count($filter=is_experienced eq true) gt 1
```
is that correct?

Reply:
@Page- no this didn't work and it actually still fails w/
> AssertionError: expected SyntaxError: Expected " ", "%20", "&", "+… { …(3) } to be an instance of Array

I guess b/c I only added the additional options to the SortProperty in the odata-parser PR :facepalm:

Reply:
Well, I guess previously this couldn't handle it but neither could odata-parser.. which explains why it didn't notice